src/config.rs

```diff
@@ -41,8 +41,33 @@ pub struct GeoIPConfiguration {
 }
 
 /// Configuration for Intelligent Peer Sharing module
-#[derive(Default, Debug, Clone, Deserialize)]
-pub struct IPSConfiguration {} //TODO(asmie): implement with IPS
+#[derive(Debug, Clone, Deserialize)]
+pub struct IPSConfiguration {
+    /// Path where peer list file will be written
+    pub peer_file_path: Option<PathBuf>,
+    /// Indicates if configuration should be taken into account
+    pub use_geolocation: bool,
+    /// True means we should prefer closer peers, false means we should prefer farther peers
+    pub use_closer_geolocation: bool,
+    /// If pref_location_closer is true, this is the maximum distance in kilometers we should prefer
+    /// closer peers. If pref_location_closer is false, this is the minimum distance in kilometers we
+    /// should prefer farther peers.
+    pub geolocation_minmax_distance_km: u32,
+    /// Indicates how many peers must be changed for each node
+    pub change_at_least: u32,
+    /// Indicates maximum peers should be changed for each node
+    pub change_no_more: u32,
+    /// Weight (importance) of the location factor (used in multi-criteria analysis)
+    pub location_weight: f64,
+    /// Weight (importance) of the degree factor (used in multi-criteria analysis)
+    pub degree_weight: f64,
+    /// Weight (importance) of the eigenvector factor (used in multi-criteria analysis)
+    pub eigenvector_weight: f64,
+    /// Weight (importance) of the betweenness factor (used in multi-criteria analysis)
+    pub betweenness_weight: f64,
+    /// Weight (importance) of the closeness factor (used in multi-criteria analysis)
+    pub closeness_weight: f64,
```

Comment:
to avoid repeating the same comment `(used in multi-criteria analysis)` 

We could consider to move these values into a mca struct:
```rust
/// Algorithm weight factors.
pub mca: MultiCriteriaAnalysisWeights;
```
now there's no need to add the `weight` keyword to every name
```rust
struct MultiCriteriaAnalysisWeights {  // or just MultiCriteriaWeights
    pub location: f64,
    pub degree_weight: f64,
    [...]
}
```

Reply:
Changed. 
As this makes also changes to configuration file _config.toml_ has been also updated.